lib/ansible/plugins/filter/core.py

```diff
@@ -342,6 +343,37 @@ def combine(*terms, **kwargs):
     return result
 
 
+SEMVER_VALID_KEYS = ['major', 'minor', 'patch', 'prerelease', 'buildmetadata']
+# This list can be generated from the SEMVER_RE expression by
+# `SEMVER_VALID_KEYS=list(SEMVER_RE.groupindex.keys())` but is expected to stay static
```

Comment:
generate it as frozenset at class level, to avoid hardcoded and redefining per invocation

Reply:
as in the version class?

Reply:
yes